### DISCUSSION TITLE
Nextcloud hangs after fresh install (maybe related to NGINX reverse proxy)

### DISCUSSION THREAD

**Discussion by cruftex** [nextcloud/all-in-one#2935] [Questions] [answered]
### Steps to reproduce
1. Fresh Ubuntu 22.04
2. NGINX Proxy Setup from documentation
3. Fresh install of NC 26 with data dir on host: /mnt/ncdata
4. Open demo office documents: works
5. Open Photos: Connection is stuck, no answer from server
6. I can login again, and see the navigation, but as soon as I open something there is no answer

### Expected behavior 

Files open, directories got listed

### Actual behavior 

Connection is stuck

### Host OS 

Ubuntu 22.04

#### Nextcloud AIO version 

Nextcloud AIO v6.2.1

#### Current channel 

latest

#### Other valuable info 

A hint where to start debugging, or whether I should try without NGINX reverse proxy first would be appreciated

**Comment by szaimen** [comment]
Hi, can you post the output of `sudo docker logs nextcloud-aio-imaginary` here?

**Comment by cruftex** [comment]
There is no log output. Here some more info:

````
root@fiji:~# docker logs nextcloud-aio-imaginary
root@fiji:~# docker exec -it nextcloud-aio-imaginary sh
~ $ netstat -l
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       
tcp        0      0 localhost:35337         0.0.0.0:*               LISTEN      
tcp        0      0 :::9000                 :::*                    LISTEN      
udp        0      0 localhost:40646         0.0.0.0:*                           
Active UNIX domain sockets (only servers)
Proto RefCnt Flags       Type       State         I-Node Path
~ $ ps
PID   USER     TIME  COMMAND
    1 nobody    0:11 imaginary -return-size -max-allowed-resolution 222.2
14657 nobody    0:00 sh
14663 nobody    0:00 ps
~ $ lsof -p 1
1       /usr/local/bin/imaginary        0       /dev/null
1       /usr/local/bin/imaginary        1       pipe:[124525]
1       /usr/local/bin/imaginary        2       pipe:[124526]
1       /usr/local/bin/imaginary        3       anon_inode:[eventpoll]
1       /usr/local/bin/imaginary        4       pipe:[126014]
1       /usr/local/bin/imaginary        5       pipe:[126014]
1       /usr/local/bin/imaginary        6       socket:[126030]
14657   /bin/busybox    0       /dev/pts/0
14657   /bin/busybox    1       /dev/pts/0
14657   /bin/busybox    2       /dev/pts/0
14657   /bin/busybox    10      /dev/tty
root@fiji:~# docker restart nextcloud-aio-imaginary
nextcloud-aio-imaginary
root@fiji:~# docker logs nextcloud-aio-imaginary
2023/07/11 08:31:59 Graceful shutdown
~ $ exit
````

I did reboot the server. After login the dashboard shows. When I click on Files the files show normally, when I click on Photos the connection hangs again.

````
root@fiji:~# free
               total        used        free      shared  buff/cache   available
Mem:         3907536     2334080      109036       90312     1464420     1214888
Swap:              0           0           0
````

I added 2GB of swap and restarted. Same problem.

Next, I disabled preview generation, via:

````
  'enable_previews' => false,
````

Now everything is reacting normally and clicking on Photos works without getting stuck.

So, something seems to be off with imaginary. What to try next?

**Comment by szaimen** [comment]
I'll move this to discussions for now since I cannot reproduce the issue.

**Comment by szaimen** [comment]
Can you disable imaginary temporarily and check if that fixes the issue for you?

**Comment by cruftex** [comment]
See my last post. I disabled "enable_preview" then everything seems to work

**Reply by szaimen** [reply]
> See my last post. I disabled "enable_preview" then everything seems to work

Yes I saw this but it is a bad workaround. Did you try if only disabling imaginary also already makes it work?

**Comment by cruftex** [comment]
> Yes I saw this but it is a bad workaround. Did you try if only disabling imaginary also already makes it work?

Ah! Sorry, I did not realise that is any different. Searching for imaginary in nextcloud and googling for disabling it did not yield any results. So I checked the config again and used common sense. I changed the config now to:

````
    6 => 'OC\\Preview\\Movie',
    7 => 'OC\\Preview\\Krita',
//    0 => 'OC\\Preview\\Imaginary',
  ),
  'enable_previews' => true,
````

Is this what I should do? With this setting, everything is "clickable" as well. No hangs.

**Reply by szaimen** [reply, marked as answer]
> Is this what I should do?

No, the easiest way would have been to disable imaginary from the AIO interface.

> With this setting, everything is "clickable" as well. No hangs.

I see. We have a [fix](https://github.com/nextcloud/all-in-one/pull/2926) scheduled for the next release which hopefully fixes the problem for you.

**Comment by cruftex** [comment]
@szaimen 
I switched off imaginary via the AIO config and restarted. Strangely, I still get hangs related to the previews.
E.g. the request to `/core/preview?fileId=38&x=250&y=250` does not get an answer.

I started digging and debugging a bit further.

What hangs now (or in general, see below) is OpenDocument preview generation.

Test 1: When I disable preview generation altogether, I can open OpenDocument files via Collabra.

Test 2: I enable preview generation and disable OpenDocument preview generation. Then, markdown and image previews are shown.

I am running out of time for today, next things to investigate:

I am running on 2 cores. The preview generation limits the parallel processes to 4 then. That means these would be exhausted when displaying the Dashboard since there are 5 open office files for demonstration. When clicking on Photos the preview requests will get stuck in the semaphore limiting the parallelism. My problem might be unrelated to Imaginary. So maybe I could switch imaginary on again, and keep OpenDocument preview generation off to test further.

The allowed parallelism depends on the cores. How many cores are you using for testing?

**Reply by szaimen** [reply]
My test server also has 2 cores.
